graylog2-web-interface/src/components/bootstrap/MenuItem.tsx

```diff
@@ -81,12 +85,19 @@ const CustomMenuItem = <T, >({ children, className, disabled, divider, eventKey,
   };
 
   if (href) {
+    const _onClickHref: React.MouseEventHandler<HTMLAnchorElement> = (isAbsoluteUrl(href) || rel || target) ? undefined
+      : (e) => {
+        history.push(href);
+        e.preventDefault();
+      };
+
     return (
       <StyledMenuItem component="a"
                       href={href}
                       target={target}
                       rel={rel}
-                      {...sharedProps}>
+                      {...sharedProps}
+                      onClick={_onClickHref}>
```

Comment:
It looks like this is not covered in the mantine `Menu` documentation, but having a look at the [mantine anchor](https://mantine.dev/core/anchor/#polymorphic-component) documentation, the recommended way seems to be using the `react-router-dom` `Link` component (or our abstraction) for the `component` prop. I tried the following props:

```
component={Link}
to={href}
```

and removed the `href` prop and it worked well, even with external links. This seems to be a bit cleaner because we can avoid the `onClick` logic including the `isAbsoluteUrl` check. Do you see any disadvantages?

Reply:
No matter which solution we choose, we should also make this change for the `Button` component.

Reply:
I just tried nesting the `Link` component and it was messing with the styling. Your idea is great, I am gonna test it this way.

Reply:
Nice. Are you sure we really need the `isAbsolute` check and different props for each case?
I tried it this way ` <StyledMenuItem component={Link} to={href} target={target} rel={rel} {...sharedProps}>` and it seems to work well for both cases.

Reply:
I noticed when I define an external link with the current solution, nothing happens on click:
![link](https://github.com/Graylog2/graylog2-server/assets/46300478/63a49f84-bd56-4df5-ad6f-df8b0a44621c)

Reply:
I used `https://www.graylog.com` in this example.

Reply:
I cleaned it up, fixing that issue too.